src/components/virtual-scroll/virtual-scroll.component.ts

```diff
@@ -321,7 +321,13 @@ export class VirtualScroll<T> implements VirtualScrollState<T> {
             const bufferLengthPx = (scrollContainer.clientHeight) * bufferLength;
 
             // Calculate the number of rendered items per row
-            const itemsPerRow = gridList ? Math.floor(scrollContainer.clientWidth / itemWidth!) : 1;
+            let itemsPerRow = gridList ? Math.floor(scrollContainer.clientWidth / itemWidth!) : 1;
+
+            // Don't reset items per row to 0 on view change as it invalidates the viewCache
+            if (itemsPerRow === 0 && this._itemsPerRow > 0) {
```

Comment:
We should be able to safely skip running all of the logic in this subscription when the list element has a width or height of 0. Instead of checking for `itemsPerRow` being 0 here, we could update the `filter` condition on line 304 to additionally check that the `clientWidth` and `clientHeight` of `scrollContainer` are both greater than 0.

This should hopefully result in the same fixed behavior and also avoid updating other state variables unnecessarily.

Reply:
I'll test it out and see if the behaviour is the same, will update the PR accordingly, thanks